iscsi/get.go

```diff
@@ -0,0 +1,314 @@
+func GetStats(rootPath string, iqnPath string) (*Stats, error) {
+	var istats Stats
+
+	istats.Name = filepath.Base(iqnPath)
+	istats.RootPath = rootPath
+	sysPath = istats.RootPath
+	matches, err := filepath.Glob(filepath.Join(iqnPath, "tpgt*"))
+	if err != nil {
+		return nil, fmt.Errorf("lio: get TPGT error %v", err)
+	}
+	istats.Tpgt = make([]TPGT, len(matches))
+
+	for ii, tpgtPath := range matches {
+		istats.Tpgt[ii].Name = filepath.Base(tpgtPath)
+		istats.Tpgt[ii].TpgtPath = tpgtPath
+		istats.Tpgt[ii].IsEnable, _ = isPathEnable(tpgtPath)
+		if istats.Tpgt[ii].IsEnable {
+			matchesLunsPath, _ := getLun(tpgtPath)
+			istats.Tpgt[ii].Luns = make([]LUN, len(matchesLunsPath))
+
+			for ll, lunPath := range matchesLunsPath {
+				backstore, objectName, typeNumber, err := getLunLinkTarget(lunPath)
+				if err != nil {
+					continue
+				}
+				istats.Tpgt[ii].Luns[ll].Name = filepath.Base(lunPath)
+				istats.Tpgt[ii].Luns[ll].LunPath = lunPath
+				istats.Tpgt[ii].Luns[ll].Backstore = backstore
+				istats.Tpgt[ii].Luns[ll].ObjectName = objectName
+				istats.Tpgt[ii].Luns[ll].TypeNumber = typeNumber
+			}
+		}
+	}
+	return &istats, nil
+}
+
+// isPathEnable is a utility function
+// check if the file "enable" contain enable message
+func isPathEnable(path string) (bool, error) {
+	var isEnable bool
+	isEnable = false
+	tmp, err := ioutil.ReadFile(filepath.Join(path, "enable"))
+	if err != nil {
+		return false, fmt.Errorf("is Path Enable error %v", err)
+	}
+	tmpNum, err := strconv.Atoi(strings.TrimSpace(string(tmp)))
+	if err != nil {
+		return false, fmt.Errorf("Atoi fail %v", err)
+	}
+	if tmpNum > 0 {
+		isEnable = true
+	}
+	return isEnable, nil
+}
+
+func getLun(tpgtPath string) (matches []string, err error) {
+	matches, err = filepath.Glob(filepath.Join(tpgtPath, "lun/lun*"))
+	if err != nil {
+		return nil, fmt.Errorf("getLun error  %v", err)
+	}
+	return matches, nil
+}
+
+func getLunLinkTarget(lunPath string) (backstoreType string,
+	objectName string, typeNumber string, err error) {
+	files, err := ioutil.ReadDir(lunPath)
+	if err != nil {
+		return "", "", "", fmt.Errorf("lio getLunLinkTarget error  %v", err)
+	}
+	for _, file := range files {
+		fileInfo, _ := os.Lstat(lunPath + "/" + file.Name())
+		if fileInfo.Mode()&os.ModeSymlink != 0 {
+			target, err := os.Readlink(lunPath + "/" + fileInfo.Name())
+			if err != nil {
+				return "", "", "", fmt.Errorf("Readlink err %v", err)
+			}
+			p1, objectName := filepath.Split(target)
+			_, typeWithNumber := filepath.Split(filepath.Clean(p1))
+
+			real_udev, err := ioutil.ReadFile(lunPath + "/" + target + "/udev_path")
+			if err != nil {
+				return "", "", "", fmt.Errorf("udev_path read file err %v", err)
+			}
+			udev_number, err := os.Readlink(strings.TrimSpace(string(real_udev)))
+
+			if err != nil {
+				return "", "", "", fmt.Errorf("udev_path readlink err %v", err)
+			}
+			tmp := strings.Split(typeWithNumber, "_")
```

Comment:
Give this a more meaningful name than `tmp`.

Reply:
done, splitTypeWithNumber it is